proto/storage/collector_runtime_config.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+
+package storage;
+
+import "storage/resource_collection.proto";
+
+option go_package = "storage";
+option java_package = "io.stackrox.proto.storage";
+
+message CollectorRuntimeConfigACK {
+  string error = 1;
+  Action action = 2;
+
+  enum Action {
+    ACK = 0;
+    NACK = 1;
+  }
+}
+
+enum RuntimeFeature {
+  PROCESSES = 0;
+  NETWORK_CONNECTIONS = 1;
+  NETWORK_ENDPOINTS = 2;
+}
+
+message Status {
+  message ProcessStatus {
+    bool enabled = 1;
+  }
+
+  message NetworkConnectionStatus {
+    bool enabled = 1;
+    bool aggregate_external = 2;
+  }
+
+  message NetworkEndpointStatus {
+    bool enabled = 2;
+    bool processes_listening_on_port = 3;
+  }
+
+  oneof status {
+    ProcessStatus process_status = 1;
+    NetworkConnectionStatus network_connection_status = 2;
+    NetworkEndpointStatus network_endpoint_status = 3;
+  }
+}
+
+message RuntimeFeatureConfig {
+  message RuntimeRule {
+    string resource_collection_name = 1;
+    Status status = 2;
+  }
+
+  RuntimeFeature feature = 1;
+  Status default_status = 2;
+  repeated RuntimeRule rules = 3;
+}
+
+message CollectorRuntimeConfig {
+  repeated RuntimeFeatureConfig runtime_feature_config = 1;
+  repeated ResourceCollection resource_collections = 2;
+}
+
+message CollectorRuntimeConfigWithCluster {
+  string cluster_name = 1;
+  CollectorRuntimeConfig config = 2;
+}
```

Comment:
Just thinking loud:

```suggestion
message RuntimeFeatureConfigs {
  repeated RuntimeFeatureConfig runtime_feature_config = 1;
  repeated ResourceCollection resource_collections = 2;
}

message CollectorRuntimeConfig {
  string cluster_name = 1;
  RuntimeFeatureConfigs config = 2;
}
```

Reply:
I think we should keep Collector in the protobuf name. If other components have runtime configuration in the future, we could wrap them up together into `RuntimeFeatureConfigs`

Reply:
Resolving as @ovalenti is on PTO.